src/Host/Protocol/Impl/RUserProfile/RUserProfileServices.cs

```diff
@@ -69,7 +75,7 @@ public class RUserProfileCreator {
             }
         }
 
-        private static async Task CreateProfileHandleUserCredentails(IUserProfileServices userProfileService, Stream stream,  CancellationToken ct, ILogger logger = null) {
+        private static async Task HandleUserCredentailsAsync(Stream stream, Func<IUserCredentials, ILogger, IUserProfileServiceResult> operation, CancellationToken ct, ILogger logger = null) {
```

Comment:
The name seems to be misleading in general - I don't see any code in the body that deals with credentials? It just deserializes request, invokes the operation, and serializes response.